zulip/zulip/__init__.py

```diff
@@ -487,6 +487,10 @@ def __init__(
         self.has_connected = False
 
         server_settings = self.get_server_settings()
+
+        if server_settings["result"] != "success":
+            raise ZulipError(server_settings["msg"])
```

Comment:
It seems to me that `get_server_settings()` itself should do this check and raise an exception—this is a case of #672.

Reply:
The work to resolve #672 seems outside of the scope of this PR since this isn't really an endpoint-specific error, but a generic one that we return ourselves. If you look at [this line](https://github.com/zulip/python-zulip-api/blob/main/zulip/zulip/__init__.py#L656), would it be better to treat this timeout exception as an `UnrecoverableNetworkError` and raise it there in `do_api_query` (just like we do for SSL errors)? That way, we can keep the scope of this PR relatively small and think about #672 separately?

Reply:
Huh that points out an error in my own phrasing of the commit message, this isn't really specific to get_server_settings, now that I think about it.

Reply:
The point here is not specific to timeout exceptions: The point is that any error that results in the `server_settings` request failing should result in an exception being raised.  It would probably be better for the "original" exception to be reraised.

I agree this is a special case of #672, though I think the right fix for that probably involves making `do_api_query` do the right thing with exceptions.  Do we think this is a reasonable tactical fix and we should just extend #672 with a note that we should remember to remove this when we fix it?

Reply:
@timabbott I just pushed some changes and they are what would represent a tactical fix we could possibly merge. Please let me know what you think! Thanks! :)